src/linear_algebra/annihilating_polynomial.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2022 Justin Thomas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Justin Thomas
+-/
+import data.polynomial
+import ring_theory.principal_ideal_domain
+import algebra.module.linear_map
+import field_theory.minpoly
+import linear_algebra
+import ring_theory.ideal.operations
+import ring_theory.polynomial_algebra
+
+/-!
+# Annihilating Ideal in general case and a specialization where the minimal polynomial generates
+
+Given a commutative ring `A` and an `A`-module `M`
+(`[comm_ring A] [add_comm_group M] [module A M]`)
+Every `A`-linear endomorphism `u: M →ₗ[A] M` defines
+an ideal (`alg_hom.annihilating_ideal u ⊆ A[X]`.
+Simply put, this is the set of polynomials `p` where
+the endomporphism `p(u): M →ₗ[A] M` is the constant 0.
+
+In the special case that `A` is a field, we use the notation `A = 𝕜`.
+Here `𝕜[X]` is a PID, so there is a polynomial `g ∈ alg_hom.annihilating_ideal u`
+which generates the ideal. We show that if this generator is
+chosen to be monic, then it is the minimal polynomial of `u`,
+as defined in `field_theory.minpoly`.
+-/
+
+variables {A M B : Type*}
+
+variables [comm_ring A] [add_comm_group M] [module A M]
+
+/-
+First we make sense of the informal notation `p(u)`.
+Think of `u` as the image of `X` in an algebra map `A[X] → module.End A M`
+and extend this to all polynomials `p(X)` using the `A` algebra structure
+on `module.End A M` (`= (M →ₗ[A] M)`).  This is given to us by `polynomial.eval`.
+Using this definition, we can define the annihilating ideal of `u` to be
+`{ p ∈ A[x] | p(u) = 0 }`. This is an ideal in `A[X]`. -/
+noncomputable def annihilating_ideal (u: M →ₗ[A] M) : ideal (polynomial A) :=
+  ring_hom.ker (polynomial.aeval u).to_ring_hom
+
+/-- It is useful to refer to ideal membership sometimes
+ and the annihilation condition other times -/
+lemma mem_annihilating_ideal_iff_aeval_0 (u: M →ₗ[A] M) (p: polynomial A)
+ : (p ∈ annihilating_ideal u) ↔ (polynomial.aeval u) p = 0 :=
+begin
+  split,
+  intros hp,
+  exact (ring_hom.mem_ker (polynomial.aeval u).to_ring_hom).2 hp,
+  intros hup0,
+  exact (ring_hom.mem_ker (polynomial.aeval u).to_ring_hom).1 hup0,
+end
```

Comment:
There's no point even entering tactic mode here, I suspect you can just write the term directly.

Reply:
Indeed, the entire proof is just
````suggestion
(ring_hom.mem_ker (polynomial.aeval u).to_ring_hom).symm
```